### PR TITLE
start of using Systems with io ops

### PR DESCRIPTION
**Pull request by anandijain**
could use a little help though.

reading back into system isn't working

### CODE REVIEW

**Comment by anandijain** [comment]
FYI I did see the serialization testset, but I figure that we will also want easy IO usage too.

**Comment by anandijain** [comment]
for a little context the usage I'm aiming for is `write("rc.jl", rc_model)` and `read("rc.jl", ODESystem)`

File: src/ModelingToolkit.jl [hunk 1]
```diff
@@ -53,6 +53,7 @@ import Symbolics: rename, get_variables!, _solve, hessian_sparsity,
 import DiffEqBase: @add_kwonly
 
 import LightGraphs: SimpleDiGraph, add_edge!
+import Base.write, Base.read
```
Comment (YingboMa) [inline src/ModelingToolkit.jl:56]: We shouldn't import it when we just overload it in one place.

File: src/ModelingToolkit.jl [hunk 2]
```diff
@@ -181,5 +182,6 @@ export build_function
 export modelingtoolkitize
 export @variables, @parameters
 export @named, @nonamespace
+export read, write
```
Comment (YingboMa) [inline src/ModelingToolkit.jl:185]: The are already in Base, so we shouldn't export it.

File: src/systems/abstractsystem.jl [hunk 3]
```diff
@@ -737,3 +737,7 @@ end
 function connect(syss...)
     connect(promote_connect_type(map(get_connection_type, syss)...), syss...)
 end
+
+# idk where to put them 
+write(io::IO, sys::T) where {T <: ModelingToolkit.AbstractODESystem} = write(io, string(toexpr(sys)))
```
Comment (YingboMa) [inline src/systems/abstractsystem.jl:742]: Just write `Base.write`.

File: src/ModelingToolkit.jl (hunk 1 above)
Comment (anandijain) [reply src/ModelingToolkit.jl:56]: What is the proper way to do this? REPL told me to import, so I did

File: src/ModelingToolkit.jl (hunk 2 above)
Comment (anandijain) [reply src/ModelingToolkit.jl:185]: Thanks

File: test/systemio.jl [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# I can't get buffer to work with read
+# io = IOBuffer() 
+
+open("rc.jl", "w") do io 
+    write(io, rc_model)
+end
+
+open("rc.jl", "r") do io 
+    ser = read(io, ODESystem)
```
Comment (YingboMa) [inline test/systemio.jl:13]: Does this even work? You are declaring `ser` in an anonymous function.

Comment (anandijain) [reply test/systemio.jl:13]: no I mentioned it doesn't. How should I do this?

File: src/ModelingToolkit.jl (hunk 1 above)
Comment (YingboMa) [reply src/ModelingToolkit.jl:56]: Just don't do this. Write `Base.write`.

File: src/systems/abstractsystem.jl (hunk 3 above)
Comment (anandijain) [reply src/systems/abstractsystem.jl:742]: kk

**Comment by anandijain** [comment]
also what file should this go in? abstractsystem.jl okay?

File: src/systems/abstractsystem.jl [hunk 5]
```diff
@@ -737,3 +737,7 @@ end
 function connect(syss...)
     connect(promote_connect_type(map(get_connection_type, syss)...), syss...)
 end
+
+# idk where to put them 
+Base.write(io::IO, sys::T) where {T <: ModelingToolkit.AbstractODESystem} = write(io, string(toexpr(sys)))
```
Comment (YingboMa) [inline src/systems/abstractsystem.jl:742]: You don't need to write `ModelingToolkit.AbstractODESystem` when this is already in `ModelingToolkit`...

Comment (anandijain) [reply src/systems/abstractsystem.jl:742]: haha kk

Comment (YingboMa) [reply src/systems/abstractsystem.jl]: Also, `sys:: AbstractSystem` is enough.

File: test/systemio.jl (hunk 4 above)
Comment (anandijain) [reply test/systemio.jl:13]: and by this, I don't mean just moving ser outside haha I mean getting `write("rc.jl", rc_model)` and `read("rc.jl", ODESystem)` to work

File: src/systems/abstractsystem.jl [hunk 6]
```diff
@@ -737,3 +737,7 @@ end
 function connect(syss...)
     connect(promote_connect_type(map(get_connection_type, syss)...), syss...)
 end
+
+# idk where to put them 
+Base.write(io::IO, sys::T) where {T <: AbstractODESystem} = write(io, string(toexpr(sys)))
```
Comment (YingboMa) [inline src/systems/abstractsystem.jl:742]: `sys::AbstractSystem`

Comment (YingboMa) [reply src/systems/abstractsystem.jl:742]: Also, do you need `string` to `write`?

Comment (anandijain) [reply src/systems/abstractsystem.jl:742]: So `toexpr` works for all Systems? Nice!

Comment (YingboMa) [reply src/systems/abstractsystem.jl:742]: No, it doesn't. But it just doesn't make sense to constraint the type in here.

**Comment by anandijain** [comment]
yeah what is the right way to do `read` cuz that looks wrong

File: test/rc.jl [hunk 7]
```diff
@@ -0,0 +1,8 @@
+begin
```
Comment (anandijain) [inline test/rc.jl:1]: ill remove this in the actual PR

**Pull request closed without merging**